lib/session/call-session.js

```diff
@@ -1790,6 +1792,17 @@ class CallSession extends Emitter {
     //this.logger.debug(`updating redis with ${JSON.stringify(this.callInfo)}`);
     this.updateCallStatus(Object.assign({}, this.callInfo.toJSON()), this.serviceUrl)
       .catch((err) => this.logger.error(err, 'redis error'));
+
+    if (this.wakeupResolver && !this.dlg && this.isCurrentCallHigherThan(CallStatus.Failed)) {
+      // Someone is waiting for call to be final
+      this.wakeupResolver({reason: 'session ended'});
+      this.wakeupResolver = null;
+    }
+  }
+
+  isCurrentCallHigherThan(target) {
```

Comment:
Wow, I don't understand what this code is doing.  Can you explain ?

Reply:
what does "higher than" mean in this case?

Reply:
For example: Call is final when callStatus is in [Faled, Busy, NoAnswer, Completed]

Here the function is has typo issue that I change name to isCurrentCallStatusHigherThan

I want to use index of each key,value pair in json to identify if call is already final